src/config.c

```diff
@@ -24,6 +24,33 @@
 #include <ctype.h>
 #include <sys/stat.h>
 
+
+const struct vpn_config invalid_cfg = {
+	.gateway_host = {'\0'},
+	.gateway_port = 0,
+	.username = {'\0'},
+	.password = {'\0'},
+	.otp = {'\0'},
+	.cookie = {'\0'},
+	.realm = {'\0'},
+	.set_routes = -1,
+	.set_dns = -1,
+	.pppd_use_peerdns = -1,
+	.use_syslog = -1,
+	.half_internet_routes = -1,
+	.pppd_log = NULL,
+	.pppd_plugin = NULL,
+	.pppd_ipparam = NULL,
+	.pppd_call = NULL,
```

Comment:
isn't `.pppd_ifname = NULL,` missing here?

Reply:
Ah, it wasn't in the default struct when I copied it, and I only fixed that over there; good catch.
I was actually thinking of removing all the 0/NULL setting here though as that is the default values; being explicit has some advantages but it's also more maintenance work